graph/config/cytoscape/cytoscape.go

```diff
@@ -105,6 +105,7 @@ type NodeData struct {
 	HasTrafficShifting    bool                `json:"hasTrafficShifting,omitempty"`    // true (vs has traffic shifting) | false
 	HasVS                 *VSInfo             `json:"hasVS,omitempty"`                 // it can be empty if there is a VS without hostnames
 	HasWorkloadEntry      []graph.WEInfo      `json:"hasWorkloadEntry,omitempty"`      // static workload entry information | empty if there are no workload entries
+	HealthData            interface{}         `json:"healthData"`                      // data to calculate health status from configurations
```

Comment:
Is there a reason we don't describe this datatype in more detail?

Reply:
I had to port some typescript code to go, so I was influenced by typescript typing system. I wanted this to be a union type and the closest that go offers is the generic `interface{}`. To remove the generic type I would need to create a struct with three fields: one field for each possible allowed type: workloadHealth, serviceHealth or appHealth. Only one field would be populated on an instance of the struct.

Reply:
Actually, there is also a fourth type which I also mentioned in the comment that I already pointed in your other comment: https://github.com/kiali/kiali/pull/4772#discussion_r822926789.